engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala

```diff
@@ -146,43 +148,35 @@ class EmbeddedDeploymentManager(modelData: ModelData, engineConfig: Config,
 
   override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] = Future.successful {
     interpreters.get(name).map { interpreterData =>
+      val stateStatus = interpreterData.scenarioInterpreter.fold(EmbeddedStateStatus.failed, _.readStatus())
       ProcessState(
         deploymentId = interpreterData.deploymentId,
-        status = toScenarioStateStatus(interpreterData.scenarioInterpreter.map(_.status())),
+        status = stateStatus,
         version = Some(interpreterData.processVersion),
         definitionManager = processStateDefinitionManager
       )
     }
   }
 
-  private def toScenarioStateStatus(taskStatusTry: Try[TaskStatus]): StateStatus = taskStatusTry match {
-    case Failure(ex) => EmbeddedStateStatus.failed(ex)
-    case Success(TaskStatus.Running) => SimpleStateStatus.Running
-    case Success(TaskStatus.DuringDeploy) => SimpleStateStatus.DuringDeploy
-    case Success(TaskStatus.Restarting) => EmbeddedStateStatus.Restarting
-    case Success(other) => throw new IllegalStateException(s"Not supporter task status: $other")
-  }
+  override def processStateDefinitionManager: ProcessStateDefinitionManager = EmbeddedProcessStateDefinitionManager
 
   override def close(): Unit = {
     interpreters.values.foreach(_.scenarioInterpreter.foreach(_.close()))
+    deploymentStrategy.close()
     logger.info("All embedded scenarios successfully closed")
   }
 
   override def test[T](name: ProcessName, canonicalProcess: CanonicalProcess, testData: TestData, variableEncoder: Any => T): Future[TestProcess.TestResults[T]] = {
-    Future{
+    Future {
       modelData.withThisAsContextClassLoader {
         val espProcess = ProcessCanonizer.uncanonizeUnsafe(canonicalProcess)
-        KafkaTransactionalScenarioInterpreter.testRunner.runTest(modelData, testData, espProcess, variableEncoder)
+        deploymentStrategy.testRunner.runTest(modelData, testData, espProcess, variableEncoder)
       }
     }
   }
 
-  private def parseScenario(canonicalProcess: CanonicalProcess): Future[EspProcess] =
-    ProcessCanonizer.uncanonize(canonicalProcess) match {
-      case Valid(a) => Future.successful(a)
-      case Invalid(e) => Future.failed(new IllegalArgumentException(s"Failed to parse scenario: $e"))
-    }
-
-  case class ScenarioInterpretationData(deploymentId: String, processVersion: ProcessVersion, scenarioInterpreter: Try[KafkaTransactionalScenarioInterpreter])
+  private case class ScenarioInterpretationData(deploymentId: String,
+                                                processVersion: ProcessVersion,
+                                                scenarioInterpreter: Try[deploymentStrategy.ScenarioInterpreter])
```

Comment:
1. I've replaced path dependant types with just `Deployment` here and other related places (`DeploymentStrategy` and concrete implementations) and it looks like everything compile. Why this complication? Why not just use polymorphism? It looks like YAGNI. Or maybe you see some purpose in veerrry near future?
2. Can you change `interpreters` to `deployments` and `ScenarioInterpretationData.scenarioInterpreter` to `ScenarioDeploymentData.deployment`?

Reply:
:+1: